scripts/gen_eval_files.lua

```diff
@@ -385,6 +391,46 @@ local function render_eval_meta(f, fun, write)
   write(render_fun_sig(funname, params))
 end
 
+--- @return table<string, vim.ExCmd>
+local function get_excmd_meta()
+  --- @type vim.ExCmd[]
+  local cmds = require('src/nvim/ex_cmds').cmds
+  local cmd_table = {} ---@type table<string, vim.ExCmd>
+  for _, value in ipairs(cmds) do
+    cmd_table[value.command] = value
+  end
+  return cmd_table
+end
+
+--- @param cmd_name string
+--- @param cmd vim.ExCmd
+--- @param write fun(line: string)
+local function render_excmd_meta(cmd_name, cmd, write)
+  assert(cmd_name == cmd.command)
+  if cmd.removed then
+    -- skip deprecated or removed (E319) commands
+    return
+  end
+
+  local function write_vimcmd_fun(name, full_name)
+    write('')
+    write(
+      ('---' .. (full_name and ('|:%s|\n---'):format(full_name) or ''))
+        .. (cmd.desc or 'TODO: Document for Ex-cmd is missing')
+    )
+
+    if name:match('[^a-zA-Z0-9]') or LUA_KEYWORDS[name] then
```

Comment:
All vim.cmd.* should have `@param ... string`.

We can also add additional metadata to `ex_cmds.lua` if we want to annotate more specific params for each command. E.g. `vim.cmd.colorscheme` takes a single optional string argument name.

Reply:
Alright, I think I could include params into the scope of this PR easily:

(Rough) Plan:

- `EX_NEEDARG`: argument required -> `(...)`; otherwise, `()`.
- Similarily for `EX_COUNT`, etc.
- `EX_EXTRA`: allow extra args after command name -> becomes `...`; otherwise it's a single arg